ModelBundle/Document/RouteDocument.php

```diff
@@ -0,0 +1,440 @@
+<?php
+
+namespace OpenOrchestra\ModelBundle\Document;
+
+use OpenOrchestra\ModelInterface\Model\RouteDocumentInterface;
+use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;
+
+/**
+ * Class RouteDocument
+ *
+ * @ODM\Document(
+ *   collection="route_document",
+ *   repositoryClass="OpenOrchestra\ModelBundle\Repository\RouteDocumentRepository"
+ * )
+ */
+class RouteDocument implements RouteDocumentInterface
+{
+    /**
+     * @var string $id
+     *
+     * @ODM\Id
+     */
+    protected $id;
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $name;
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $pattern;
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $host;
+
+    /**
+     * @var array
+     *
+     * @ODM\Field(type="raw")
+     */
+    protected $defaults = array();
+
+    /**
+     * @var array
+     *
+     * @ODM\Field(type="raw")
+     */
+    protected $requirements = array();
+
+    /**
+     * @var array
+     *
+     * @ODM\Field(type="raw")
+     */
+    protected $options = array();
+
+    /**
+     * @var string|array
+     *
+     * @ODM\Field(type="raw")
+     */
+    protected $schemes;
+
+    /**
+     * @var string|array
+     *
+     * @ODM\Field(type="raw")
+     */
+    protected $methods;
+
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $token0;
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $token1;
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $token2;
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $token3;
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $token4;
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $token5;
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $token6;
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $token7;
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $token8;
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $token9;
+
+    /**
+     * @var string
+     *
+     * @ODM\Field(type="string")
+     */
+    protected $token10;
```

Comment:
Why not use a token collection instead of 10 scalar attributes ? It's a hard limitation